core/src/main/java/jenkins/util/VirtualFile.java

```diff
@@ -546,7 +546,11 @@ public static VirtualFile forFile(final File f) {
 
         private boolean isIllegalSymlink() {
             try {
-                return !this.isDescendant("");
+                String myPath = f.toPath().toRealPath().toString();
```

Comment:
Do you need to use the relaxed version here, too?

Reply:
I rollback my previous changes here to bring back the previous behavior. In case of exception, instead of putting the relaxed version, it accepted as non illegal. That's not the same behavior as the relaxed method.